website/project/model.py

```diff
@@ -146,13 +145,31 @@ class MetaData(GuidStoredObject):
     date_created = fields.DateTimeField(auto_now_add=datetime.datetime.utcnow)
     date_modified = fields.DateTimeField(auto_now=datetime.datetime.utcnow)
 
+class Commentable(object):
 
-class Comment(GuidStoredObject, SpamMixin):
+    @property
+    def target_type(self):
+        raise NotImplementedError
+
+    @property
+    def root_target_page(self):
+        raise NotImplementedError
+
+    @property
+    def is_deleted(self):
+        raise NotImplementedError
+
+    def belongs_to_node(self, node):
```

Comment:
Could also consider adding default behavior here, as the same behavior is given in three of four overrides.